daemon/swap_daemon_test.go

```diff
@@ -177,6 +181,16 @@ func createTestConf(t *testing.T, ethKey *ecdsa.PrivateKey) *SwapdConfig {
 
 func launchDaemons(t *testing.T, timeout time.Duration, configs ...*SwapdConfig) context.Context {
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	swapdCtx, swapdCancel := context.WithCancel(ctx)
+
+	go func() { // Handle signals
+		sigc := make(chan os.Signal, 1)
+		signal.Ignore(syscall.SIGHUP)
+		signal.Notify(sigc, syscall.SIGINT, syscall.SIGTERM)
+
+		<-sigc
+		swapdCancel()
+	}()
```

Comment:
The daemon running in tests didn't have a signal handler, added a simple one here (for the shutdown test).

If we end up keeping the signal method to shutdown the server, it would be better to use the actual signal handler used in swapd, I didn't do that here because it caused circular dependencies errors, but we could work on that if you want.

Reply:
Yeah, I usually keep signal handlers in the `main` package of my daemons.  For unit tests, we run multiple swapd instances in the same linux process. Check out my other comment. I think I found a solution that is only minor changes from the work that you've already done and doesn't send a signal.